doc/api/esm.md

```diff
@@ -1124,6 +1124,21 @@ _isImports_, _conditions_)
 >    1. Throw an _Invalid Package Configuration_ error.
 > 4. Return the parsed JSON source of the file at _pjsonURL_.
 
+**CONTAINS\_MODULE\_SYNTAX**(_source_)
+
+> 1. Parse _source_ as a CommonJS module.
+> 2. If the parse is successful, return **false**.
+> 3. Else inspect the error message thrown. Return **true** if it is a syntax
+>    error thrown by any of the following:
+>    * `import` statement (static only, _not_ dynamic `import()`)
+>    * `export` statement
+>    * `import.meta`
+>    * `await` at the top level
+>    * A lexical redeclaration of any of the CommonJS wrapper variables
+>      (`require`, `exports`, `module`, `__filename`, `__dirname`) at the top
+>      level
```

Comment:
```suggestion
>    * A static `import` statement.
>    * An `export` statement.
>    * Usage of the `import.meta` meta property.
>    * An `await` expression at the top level.
>    * A lexical redeclaration of any of the CommonJS wrapper variables
>      (`require`, `exports`, `module`, `__filename`, `__dirname`) at the top
>      level.
```

Reply:
I think it's important to clarify that this is only for import statements and not expressions.

Reply:
Note that `[[HasTLA]]` is already an ECMAScript module field - https://tc39.es/ecma262/#sec-cyclic-module-records.